### PR TITLE
RR-116 - Service method and mapper

### PR DESCRIPTION
**Pull request by nathanrussell-moj-digital**
This PR adds the basic service method and mapper to get functional skills

### CODE REVIEW

File: server/@types/viewModels/index.d.ts [hunk 1]
```diff
@@ -30,4 +30,23 @@ declare module 'viewModels' {
...
+    assessments: Array<Assessment>
+  }
+
+  /**
+   * An Assessment of a single functional skill.
+   */
+  export interface Assessment {
+    prisonId: string
+    prisonName: string
+    type?: 'ENGLISH' | 'MATHS' | 'DIGITAL_LITERACY'
```
Comment (nathanrussell-moj-digital) [inline server/@types/viewModels/index.d.ts:48]: I wanted to do a proper enum here, but couldn't work out how to get it working within the `@types` folder thingy

File: server/routes/overview/mappers/functionalSkillsMapper.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    prisonName,
+    type: toAssessmentTypeOrNull(assessment.qualificationType),
+    grade: assessment.qualificationGrade,
+    assessmentDate: dateOrNull(assessment.assessmentDate),
+  } as Assessment
+}
+
+const dateOrNull = (value: string): Date | undefined => {
+  return value ? moment(value, true).toDate() : undefined
+}
```
Comment (mattwills-moj-digital) [inline server/routes/overview/mappers/functionalSkillsMapper.ts:28]: Looks familiar 🤔

File: server/services/curiousService.ts [hunk 3]
```diff
@@ -32,4 +27,29 @@ export default class CuriousService {
...
+  async getPrisonerFunctionalSkills(prisonNumber: string, username: string): Promise<FunctionalSkills> {
+    const systemToken = await this.hmppsAuthClient.getSystemClientToken(username)
+
+    try {
+      const learnerProfiles = await this.getLearnerProfile(prisonNumber, systemToken)
+      return toFunctionalSkills(learnerProfiles)
+    } catch (error) {
+      logger.info(error)
+      if (error.code === 404) {
+        // TODO - we need to check if this is right, but Curious has been unavailable
```
Comment (mattwills-moj-digital) [inline server/services/curiousService.ts:40]: 😄

**Review by mattwills-moj-digital — APPROVED** [review]
LGTM

**Pull request merged**